### ISSUE TITLE
Unable to mutate certain properties of Objective-C objects via protocol (polymorphically)

### ISSUE THREAD

**Issue by mike-at-honor** [apple/swift#61687]
**Describe the bug**
Attempting to assign a property declared on a protocol on an Objective-C, via protocol conformance, will unexpectedly trigger a "cannot assign to property". This seems to impact struct properties - not object references.

**Steps To Reproduce**
Copy the code exactly below and try to compile:
```
    let textInput: UITextInput = UITextView()
    textInput.returnKeyType = .go // ERROR: Cannot assign to property: 'textIput' is a 'let' constant. Change 'let' to 'var' to make it mutable
    textInput.inputDelegate = nil // OK
```
Strangely, making it a `var` doesn't fix the issue, but results in a different error:
```
    var textInput2: UITextInput = UITextView()
    textInput2.returnKeyType = .go // ERROR: Cannot assign to property: 'textInput2' is immutable
```

**Expected behavior**
One would expect to be able to mutate these types of properties - it's a regression in the latest Swift, as far as I can tell.

**Environment (please fill out the following information)**
 - OS: macOS 12.6
 - Xcode Version: 14.1 RC

**Comment by mike-at-honor** [comment]
TIL.  Thanks for clarifying.  For my education - Is there a documented reason why Swift does not support this behavior, or is it just a low priority deficiency?  Albeit probably rare, any code leveraging a protocol like this would be unable to be rewritten from Objective-C into Swift.

**Comment by AnthonyLatsis** [comment]
> or is it just a low priority deficiency

Yeah, we just never got around to modeling it, which is a language design problem among other things. Is it problematic to avoid the abstraction in your case?
